### PR TITLE
Download -omics variants

### LINKED ISSUE #4701: Download DROP results from Scout
**Is your feature request related to a problem in the current program to new available techology or software? Please describe and add links/citations if appropriate.**
It would be great if the results from DROP could be downloaded from Scout so that they could be saved in LIMS if necessary

**Describe the solution you'd like**
A downloadable file containing DROP results

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #4701 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

File: scout/server/blueprints/omics_variants/controllers.py [hunk 1]
```diff
@@ -48,3 +58,62 @@ def decorate_omics_variant(store, institute_obj, case_obj, omics_variant_obj):
...
+        for line in lines:
+            yield line + "\n"
+
+    DOCUMENT_HEADER = [
+        "Gene",
+        "Gene annotation",
+        "category",
+        "sub-category",
+        "Potential impact",
+        "Value",
```
Comment (dnil) [inline scout/server/blueprints/omics_variants/controllers.py:77]: We will likely want to split this one into `delta_psi` and `l2fc`, for clarity and ease of handling - even if the `sub_category` column does show what gives.

File: scout/server/blueprints/omics_variants/controllers.py [hunk 2]
```diff
@@ -48,3 +58,62 @@ def decorate_omics_variant(store, institute_obj, case_obj, omics_variant_obj):
...
+
+    def generate(header, lines):
+        yield header + "\n"
+        for line in lines:
+            yield line + "\n"
+
+    DOCUMENT_HEADER = [
+        "Gene",
+        "Gene annotation",
+        "category",
```
Comment (dnil) [inline scout/server/blueprints/omics_variants/controllers.py:74]:
```suggestion
        "Category",
```

File: tests/server/blueprints/variants/test_variants_views.py [hunk 3]
```diff
@@ -489,7 +487,7 @@ def test_sv_cancer_variants(app, institute_obj, cancer_case_obj):
 
 
 def test_filter_export_cancer_variants(app, institute_obj, case_obj):
-    """Test the variant export functionaliy in cancer_variants page"""
+    """Test the variant export functionality in cancer_variants page"""
```
Comment (dnil) [inline tests/server/blueprints/variants/test_variants_views.py:490]: 👍

File: scout/server/blueprints/omics_variants/controllers.py [hunk 4]
```diff
@@ -48,3 +58,62 @@ def decorate_omics_variant(store, institute_obj, case_obj, omics_variant_obj):
...
+        "sub-category",
```
Comment (dnil) [inline scout/server/blueprints/omics_variants/controllers.py:75]:
```suggestion
        "Sub-category",
```

File: scout/server/blueprints/omics_variants/controllers.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/omics_variants/controllers.py:77]: > We will likely want to split this one into `delta_psi` and `l2fc`, for clarity and ease of handling - even if the `sub_category` column does show what gives.

Ah OK, good idea!

File: scout/server/blueprints/omics_variants/controllers.py [hunk 5]
```diff
@@ -48,3 +58,62 @@ def decorate_omics_variant(store, institute_obj, case_obj, omics_variant_obj):
...
+        "Potential impact",
+        "Value",
+        "P-value",
+        "Fold-change",
```
Comment (dnil) [inline scout/server/blueprints/omics_variants/controllers.py:79]: 👍 I'm sure there will be more other columns of interest as well that are not on the web page, but this is a good start of the process!

**Review by dnil — APPROVED** [review]
👍 Looks good, just some minor changes.

File: scout/server/blueprints/omics_variants/controllers.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/omics_variants/controllers.py:77]: Fixed like this now:

image

**Pull request merged**